tests/test_url_handler.py

```diff
@@ -0,0 +1,696 @@
+        url_handler.save_attribute(None, None, {'url_value': ''}, [], '', '',)
+    assert models.URL.objects.all().count() == 0
+
+
+def test_surt_exists_when_surt_exists():
+    system_nominator = models.Nominator.objects.get(id=settings.SYSTEM_NOMINATOR_ID)
+    url = factories.SURTFactory()
+    assert url_handler.surt_exists(url.url_project, system_nominator, url.entity) is True
+
+
+def test_surt_exists_when_surt_does_not_exist():
+    system_nominator = models.Nominator.objects.get(id=settings.SYSTEM_NOMINATOR_ID)
+    project = factories.ProjectFactory()
+    url = 'http://example.com'
+
+    assert len(models.URL.objects.all()) == 0
+    assert url_handler.surt_exists(project, system_nominator, url) is True
+    assert len(models.URL.objects.all()) == 1
+
+
+def test_surt_exists_when_surt_cannot_be_created():
+    system_nominator = models.Nominator.objects.get(id=settings.SYSTEM_NOMINATOR_ID)
+    project = factories.ProjectFactory()
+    url = None
+
+    with pytest.raises(http.Http404):
+        url_handler.surt_exists(project, system_nominator, url)
+
+
+@pytest.mark.parametrize('url, expected', [
+    ('www.example.com', 'http://www.example.com'),
+    ('   http://www.example.com   ', 'http://www.example.com')
+])
+def test_url_formatter(url, expected):
+    assert url_handler.url_formatter(url) == expected
+
+
+@pytest.mark.parametrize('url, preserveCase, expected', [
+    # Documentation on SURTs is inconsistent about whether a comma
+    # should come before the port or not. The assumption here is
+    # that it should.
+    ('http://userinfo@domain.tld:80/path?query#fragment', False,
+        'http://(tld,domain,:80@userinfo)/path?query#fragment'),
+    ('http://www.example.com', False, 'http://(com,example,www,)'),
+    ('https://www.example.com', False, 'http://(com,example,www,)'),
+    ('www.example.com', False, 'http://(com,example,www,)'),
+    ('http://www.eXaMple.cOm', True, 'http://(cOm,eXaMple,www,)'),
+    ('Not a URL.', False, ''),
+    ('1.2.3.4:80/examples', False, 'http://(1.2.3.4:80)/examples'),
+])
+def test_surtize(url, preserveCase, expected):
+    assert url_handler.surtize(url, preserveCase=preserveCase) == expected
+
+
+def test_appendToSurt():
+    match_obj = re.search(r'(World!)', 'The End of The World!')
+    groupnum = 1
+    surt = 'Hello, '
+    expected = 'Hello, World!'
+
+    assert url_handler.appendToSurt(match_obj, groupnum, surt) == expected
+
+
+@pytest.mark.parametrize('uri, expected', [
+    ('http://www.example.com', 'http://www.example.com'),
+    ('www.example.com', 'http://www.example.com'),
+    (':.', ':.'),
+    ('.:', 'http://.:')
+])
+def test_addImpliedHttpIfNecessary(uri, expected):
+    assert url_handler.addImpliedHttpIfNecessary(uri) == expected
+
+
+@pytest.mark.parametrize('root, text, id', [
+    ('com,', '<a href="surt/http://(com,example">com,example</a>', 'com,example,'),
+    ('', 'com', 'com,')
+])
+def test_create_json_browse(root, text, id):
```

Comment:
yeah I might change it back and add some prefix or suffix, since `id` is actually the keyword being used for the value in the dict.